server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -114,4 +118,15 @@ public Set<String> listAll(Iterable<String> path) throws IOException {
     public Set<String> listFolders(Iterable<String> path) throws IOException {
         return blobStore.blobContainer((BlobPath) path).children().keySet();
     }
+
+    @Override
+    public void listFoldersAsync(String threadpoolName, Iterable<String> path, ActionListener<Set<String>> listener) {
```

Comment:
Would it help if we pass the `ExecutorService` instead of threadpoolName? 
One thing threadpoolName ensures is that we mandate that the execution has to be done in one of the threadpools that OpenSearch manages.
On the other hand, it can be a bit stricter in places where we are writing tests.

Reply:
I think it makes sense to use threadpool name or type. This way it is easy to perform any validation or assertion that we may want to put in future. We can pass list of allowable threadpool names in the constructor of this class. Passing executor service would make it difficult to do any kind of assertions/validations.